src/app/zap-templates/templates/app/cluster-objects-src.zapt

```diff
@@ -58,6 +58,11 @@ CHIP_ERROR DecodableType::Decode(TLV::TLVReader &reader) {
         {
             {{#zcl_command_arguments}}
             case to_underlying(Fields::k{{asUpperCamelCase label}}):
+#ifdef CHIP_USE_ENUM_CLASS_FOR_IM_ENUM
```

Comment:
No, this is wrong.  CHIP_USE_ENUM_CLASS_FOR_IM_ENUM is never defined in practice.

What we need to do is to generate different bodies for the Check functions depending on whether it's a strong enum and whether CHIP_USE_ENUM_CLASS_FOR_IM_ENUM is defined....

Reply:
updated, thanks